pandas/_libs/tslibs/parsing.pyx

```diff
@@ -44,6 +51,95 @@ cdef:
     set _not_datelike_strings = {'a', 'A', 'm', 'M', 'p', 'P', 't', 'T'}
 
 # ----------------------------------------------------------------------
+cdef:
+    const char* delimiters = " /-\\."
+    int MAX_DAYS_IN_MONTH = 31, MAX_MONTH = 12
+
+
+cdef bint _is_not_delimiter(const char ch):
+    return strchr(delimiters, ch) == NULL
+
+
+cdef inline int _parse_2digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 10
+    result += getdigit_ascii(s[1], -100) * 1
+    return result
+
+
+cdef inline int _parse_4digit(const char* s):
+    cdef int result = 0
+    result += getdigit_ascii(s[0], -10) * 1000
+    result += getdigit_ascii(s[1], -100) * 100
+    result += getdigit_ascii(s[2], -1000) * 10
+    result += getdigit_ascii(s[3], -10000) * 1
+    return result
+
+
+cdef inline object _parse_delimited_date(object date_string, bint dayfirst):
+    """
+    Parse special cases of dates: MM/DD/YYYY, DD/MM/YYYY, MM/YYYY
+
+    Note
+    ----
+    For MM/DD/YYYY, DD/MM/YYYY: delimiter can be a space or one of ./\-
```

Comment:
how can you distinguish between ambiguous mm/dd/yyyy and dd/mm/yyyy? do they both work?

Reply:
no you are missing the point, its not about a doc-string change what does this do in this case? it should refuse to parse

Reply:
I just thought it was not very obvious from the code and therefore added to doc-string.
We do what `dateutil` would do in this case.

Reply:
We're actually keeping true to previous Pandas behaviour - there are multiple places which parsed the date like this, and none reported whether date was dayfirst or not...